cmd/argocd/commands/app.go

```diff
@@ -1212,6 +1212,12 @@ type objKeyLiveTarget struct {
 	target *unstructured.Unstructured
 }
 
+type objKeyRevisionTarget struct {
```

Comment:
Becasue the objects are of the same type, I think you can use `objKeyRevisionTarget` without creating a new type. The `live` property can contain the desired live manifest instead of the kubernetes live manifest.

Reply:
Okay, so I just have to write different grouping logic and will have to use `objKeyLiveTarget` to store desired and revision state right?

Reply:
Implemented this.